alf/metrics/metrics.py

```diff
@@ -266,6 +283,114 @@ def _extract_metric_values(self, time_step):
             return [reward[..., i] for i in range(reward.shape[-1])]
 
 
+@gin.configurable
+class AverageDiscountedReturnMetric(AverageEpisodicSumMetric):
+    """Metric for computing the average discounted episodic return.
+    It is calculated according to the following formula:
+
+    .. math::
+        \begin{array}{ll}
+            R &= r_1 + (1 + gamma) r_2 + (1 + gamma + gamma^2) r_3 + \cdots \\
+            &= \frac{1}{L}\sum_{l=1}^L \sum_{k=0}^{l-1} \gamma^k r_l,
```

Comment:
Do we want to average over the ending steps or the beginning steps? If this quantity is to be compared to the value estimation, then we want to average over the beginning steps. In the episodic setting:

\frac{1}{L}\sum_{l=1}^L \sum_{k=l}^{L} \gamma^{k-l} r_k,